### ISSUE TITLE
Inconsistent `BITCOUNT`: BITCOUNT against test vector #3

### ISSUE THREAD

**Issue by lucifercr07** [DiceDB/dice#660]
## Steps to reproduce

Run the commands mentioned in the test on Line 68 in the file https://github.com/AshwinKul28/dice-tests/blob/main/tcltests/unit/bitpos.tcl#L68.

## Expected output

The expected output when the above set of commands when run on Redis

```
Expected [r bitcount str 0 -1 bit]
```

## Observed output

The observed output when the above set of commands when run on DiceDB

```
8 (context: type eval line 5 cmd {assert {[r bitcount str 0 -1 bit]
```

The steps to run the test cases are mentioned in the README of the [dice-tests repository](https://github.com/AshwinKul28/dice-tests).

## Expectations for resolution

This issue will be considered resolved when the following things are done:

1. Changes in the [`dice`](https://github.com/dicedb/dice) repository code to meet the expected behavior.
1. Successful run of the tcl test behavior.

You can find the tests under the `tests` directory of the [`dice`](https://github.com/dicedb/dice) repository and the steps to run are in the [README file](https://github.com/dicedb/dice). Refer to the following links to set up DiceDB and Redis 7.2.5 locally

- [Setup DiceDB locally](https://github.com/dicedb/dice)
- [Setup Redis 7.2.5 locally](https://gist.github.com/arpitbbhayani/94aedf279349303ed7394197976b6843)
- [Setup DiceDB CLI](https://github.com/dicedb/dice)

**Comment by Nijin-P-S** [comment]
Hi @lucifercr07 , can you assign this task to me?

**Comment by lucifercr07** [comment]
@Nijin-P-S assigned, thanks for contributing.

**Comment by Nijin-P-S** [comment]
Hi @lucifercr07 ,

I wanted to reach out regarding the issues I was assigned. Despite my efforts, I’m finding it challenging to make progress on them. Additionally, I have a busy period at my full-time job coming up and won’t be able to dedicate the time needed for the next couple of weeks.

I think it would be best for me to unassign these issues so someone else can take them on. Thank you for understanding, and I hope to contribute again in the future!

**Comment by VedWhat** [comment]
@lucifercr07 I think we should icebox this issue as this is not a bitpos issue however a problem on how we're storing strings/runes. 

Similar thing for issue #661 #662 

Let's revisit this once the store refactoring goes through. ref: " We will tackle this as part of Store Refactoring when we move to storing strings and byte arrays both as byte arrays and store the native types in their original form". cc: @pratikpandey21 @soumya-codes 

Lmk what you think

**Comment by jainbhavya53** [comment]
Hi @lucifercr07 @pratikpandey21 
Can I take a shot at it ? I have worked on `BITPOS` command bug, so I have a fair idea regarding bit operations.

Please assign this issue to me, if it makes sense .

Thanks

**Comment by jainbhavya53** [comment]
Hi @arpitbbhayani  
I am still working on it, will send out a PR soon.
Thanks

**Comment by jainbhavya53** [comment]
With latest master, all `BITCOUNT` tests pass.

```
Testing unit/auth
[1/6 done]: unit/auth (0 seconds)
Testing unit/bitpos
[ok]: BITCOUNT returns 0 against non existing key (0 ms)
[ok]: BITCOUNT returns 0 with out of range indexes (1 ms)
[ok]: BITCOUNT returns 0 with negative indexes where start > end (1 ms)
[ok]: BITCOUNT against test vector #1 (0 ms)
[ok]: BITCOUNT against test vector #2 (0 ms)
[ok]: BITCOUNT against test vector #3 (1 ms)
[ok]: BITCOUNT against test vector #4 (1 ms)
[ok]: BITCOUNT against test vector #5 (0 ms)
[ok]: BITCOUNT fuzzing without start/end (211 ms)
[ok]: BITCOUNT with start, end (2 ms)
[ok]: BITCOUNT syntax error #1 (1 ms)
[ok]: BITCOUNT syntax error #2 (0 ms)
[ok]: BITCOUNT regression test for github issue #582 (0 ms)
[ok]: BITCOUNT misaligned prefix (1 ms)
[ok]: BITCOUNT misaligned prefix + full words + remainder (1 ms)
[ok]: BITOP NOT (empty string) (0 ms)
[ok]: BITOP NOT (known string) (0 ms)
[ok]: BITOP where dest and target are the same key (0 ms)
[ok]: BITOP AND|OR|XOR don't change the string with single input key (1 ms)
[ok]: BITOP missing key is considered a stream of zero (0 ms)
[ok]: BITOP shorter keys are zero-padded to the key with max length (1 ms)
```

@lucifercr07 Maybe we can close this as its no longer an issue. Seems like its fixed by https://github.com/DiceDB/dice/pull/699.

Also, we can close all related issues as well ?
https://github.com/DiceDB/dice/issues/661
https://github.com/DiceDB/dice/issues/662
https://github.com/DiceDB/dice/issues/664
https://github.com/DiceDB/dice/issues/665

Thanks

**Comment by lucifercr07** [comment]
Closing, fixed as part of #699
